apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx

```diff
@@ -0,0 +1,25 @@
+import { ContentTypeValue } from 'types';
+import { useSDK } from '@contentful/react-apps-toolkit';
+import { SidebarExtensionSDK } from '@contentful/app-sdk';
+import useGetFieldValue from '../useGetFieldValue';
+
+export const useSidebarSlug = (slugFieldInfo: ContentTypeValue) => {
+  const sdk = useSDK<SidebarExtensionSDK>();
+
+  const { slugField, urlPrefix } = slugFieldInfo;
+  const slugFieldValue = useGetFieldValue(slugField);
+
+  const reportSlug = `${urlPrefix || ''}${slugFieldValue || ''}`;
+  const slugFieldIsConfigured = Boolean(slugField);
+  const contentTypeHasSlugField = slugField in sdk.entry.fields;
+  const isPublished = Boolean(sdk.entry.getSys().publishedAt);
```

Comment:
currently working on implementing a way to listen for changes in the 'published' status of the entry. Might add to a follow-up PR if it is a significant change.